pkg/controller/application/application_controller_utils.go

```diff
@@ -30,7 +30,7 @@ func (c *Controller) createReleaseForApplication(app *shipper.Application, relea
 			Labels: map[string]string{
 				shipper.ReleaseLabel:                releaseName,
 				shipper.AppLabel:                    app.Name,
-				shipper.ReleaseEnvironmentHashLabel: hashReleaseEnvironment(app.Spec.Template),
+				shipper.ReleaseEnvironmentHashLabel: hashReleaseEnvironment(app.Spec.Template), // TODO check logic
```

Comment:
It does look like `hashReleaseEnvironment` still does the right thing, did you have something specific in mind?